src/aideck/loaders/factory.py

```
@@ -0,0 +1,837 @@
+        Raises:
+            ValueError: If no suitable loader is found
+            ValueError: If configuration is invalid
+            FileNotFoundError: If source is a file path that doesn't exist
+        """
+        try:
+            if not source:
+                raise ValueError(f"Source cannot be None or empty")
+            
+            if not source.startswith(('http://', 'https://', 'ftp://')) and not source.startswith('{') and not source.startswith('<') and not source.startswith('#'):
+                if not os.path.exists(source):
+                    raise FileNotFoundError(f"Source file does not exist: {source}")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)